replay-test/run.ts

```diff
@@ -0,0 +1,287 @@
+// Test runner for recording node scripts and replaying the resulting recordings.
+
+// Using require() here to workaround typescript errors.
+import * as fs from "fs";
+import * as path from "path";
+import { spawnSync, SpawnSyncReturns } from "child_process";
+import { TestManifest, NodeTestIgnoreList } from "./manifest";
+import { listAllRecordings, uploadRecording } from "@recordreplay/recordings-cli";
+import ProtocolClient from "./client";
+
+const Usage = `
+Usage: node run <options>
+Options:
+  --node <path>: Specify the path to the node binary to use to record.
+  --api-key <key>: Specify the API key to use for recording and replaying.
+  --run-suite: Run the default replay test suite.
+  --run-random-tests <count>: Run random tests from node's regular test suite.
+  --run-pattern <pattern>: Run all tests matching a pattern.
+  --server <address>: Set server to connect to (default wss://dispatch.replay.io).
+`;
+
+function bailout(message) {
+  console.log(message);
+  process.exit(1);
+}
+
+if (process.argv.length == 2) {
+  bailout(Usage);
+}
+
+let gNodePath: string;
+let gAPIKey: string;
+let gRunSuite: boolean;
+let gRunRandomTests: number;
+let gRunPattern: string;
+let gDispatchAddress = "wss://dispatch.replay.io";
+
+for (let i = 2; i < process.argv.length; i++) {
+  switch (process.argv[i]) {
+    case "--node":
+      gNodePath = path.resolve(process.cwd(), process.argv[++i]);
+      break;
+    case "--api-key":
+      gAPIKey = process.argv[++i];
+      break;
+    case "--run-suite":
+      gRunSuite = true;
+      break;
+    case "--run-random-tests":
```

Comment:
In what use cases would we want to use this `--run-random-tests` mode? It seems it'd be difficult to reproduce issues and determine whether a problem is fixed if this was used in CI, since we wouldn't run the same set of tests between runs.

Reply:
Running a random subset of tests is essentially an experiment.  Our CI for browser changes has a list of web sites it can load and see if they can be recorded and replayed successfully, but we don't have any equivalent for node.  The simplest alternative is to use node's existing test suite, but node has several thousand tests and rather than record a fixed subset of them it seemed preferable to select some at random to record.

Once this is, we'll be able to start testing node while the test runners are otherwise idle, mixing it in with the idle testing we currently do for gecko by recording/replaying node's tests at random.  This will surface problems with the node port without having to dedicate several hours at a time to going through the entire test suite and recording/replaying all of the tests.